modules/040-node-manager/images/bashible-apiserver/pkg/template/context_builder.go

```diff
@@ -495,7 +502,9 @@ type registryInputData struct {
 
 type dockerCfg struct {
 	Auths map[string]struct {
-		Auth string `json:"auth"`
+		Auth     *string `json:"auth"`
+		Username *string `json:"username"`
+		Password *string `json:"password"`
```

Comment:
Honestly, I don't see the need of pointers here. You don't need 3 states, only fallback. You can always compare with an empty string and fallback if true. It's not critical, so you can hold this but this gives us extra gc workload without a reason.

We can wait for @RomanenkoDenys opinion

Reply:
I think this is not important, but this struct is copy of struct from [dhctl/pkg/config/meta.go](https://github.com/deckhouse/deckhouse/pull/1515/files#diff-4b6226a078c0950de166cd15009acfea9b56a97a8264f43eca39cc72b2703c91), so we can leave this as is or change it in two places.

Reply:
Ok. I checked that empty string is normal case for checking existing auth string. I removed pointers.